website/blog/2024-03-18-run-ferretdb-securely-using-openziti.md

```diff
@@ -0,0 +1,169 @@
+
+Let's break down the key components:
+
+- Ziti Controller: Manages the Ziti network, identities, and policies.
+- Ziti Edge Router: Handles encrypted traffic between Ziti clients and services.
+- Ziti Console: Provides a web interface for managing the Ziti network.
+
+FerretDB connects to PostgreSQL and is configured to run within the same Ziti network.
+
+### Initialize Docker Environment
+
+Using the project name 'pg' for Postgres, start the Docker Compose environment:
+
+```sh
+docker compose -p pg up
+```
+
+This command sets up several services, including `ziti-controller`, `ziti-edge-router`, `ziti-console`, `postgres-db`, and `ferretdb`, as specified in the Docker Compose YAML.
+
+Run `docker ps` to show that Postgres and FerretDB are not exposed (`5432`/`27017`):
+
+### Testing your network connection
+
+To verify the security and functionality of your setup, follow these steps to test network connectivity between the Ziti components and ensure they're correctly configured.
+
+To begin, access the running Ziti Controller container using the following command:
+
+```sh
+docker exec -it pg-ziti-controller-1 bash
+```
+
+This should take you into the Ziti CLI.
+Authenticate using the `zitiLogin` alias:
+
+```sh
+zitiLogin
+```
+
+Test edge routers online:
+
+Verify that all edge routers are online and properly registered with the Ziti Controller:
+
+```sh
+ziti edge list edge-routers
+```
+
+You should see your ziti-edge-router listed and marked as ONLINE.
+
+Test edge router identities:
+
+Each edge router should have an associated identity within the Ziti network.
+Check these identities:
+
+```sh
+ziti edge list identities
+```
+
+This command lists all registered identities, including those for your routers.
+
+Test network connectivity:
+
+Let's ensure the Ziti Controller and a Ziti Edge Router can communicate over the network.
+
+Use ping from within the controller container to verify connectivity to the `ziti-edge-router`:
+
+```sh
+$ ping ziti-edge-router -c 1
+PING ziti-edge-router (172.26.0.6): 56 data bytes
+64 bytes from 172.26.0.6: icmp_seq=0 ttl=64 time=0.562 ms
+--- ziti-edge-router ping statistics ---
+1 packets transmitted, 1 packets received, 0% packet loss
+round-trip min/avg/max/stddev = 0.562/0.562/0.562/0.000 ms
+```
+
+These tests ensure that your Docker network settings allow for proper communication paths between the Ziti Controller and the Ziti Edge Router, ensuring that your FerretDB setup with OpenZiti operates more efficiently and securely.
+
+### Connecting to FerretDB
+
+Once your services are up and running, you can connect to FerretDB using the MongoDB shell (`mongosh`) over the secure network established by OpenZiti.
+
+```sh
+docker run -it --rm --network pg_ziti mongo mongosh "mongodb://postgres:postgres@pg-ferretdb-1:27017/ferretdb?authMechanism=PLAIN"
```

Comment:
I think it is better to add mongo service to Docker Compose configuration to use `docker compose run` and avoid `pg_` prefix everywhere